ts/components/messages/MessageAgenda.tsx

```diff
@@ -252,82 +242,148 @@ const generateItemLayouts = (sections: Sections) => {
 
 const ItemSeparatorComponent = () => <View style={styles.itemSeparator} />;
 
-const FakeItemComponent = (
-  <View style={styles.itemEmptyWrapper}>
-    <Text style={styles.itemEmptyText}>{I18n.t("reminders.emptyMonth")}</Text>
+const PlaceholderLine = () => (
+  <Placeholder.Line
+    textSize={customVariables.fontSizeBase}
+    color={customVariables.shineColor}
+    width={"75%"}
+    animate={"shine"}
+  />
+);
+
+const MessageItemPlaceholder = (
+  <View style={[styles.itemLoadingContainer]}>
+    <View style={styles.itemLoadingHeaderWrapper}>
+      <View style={styles.itemLoadingHeaderCenter}>
+        <Placeholder.Paragraph
+          textSize={customVariables.fontSizeBase}
+          color={customVariables.shineColor}
+          lineNumber={2}
+          lineSpacing={5}
+          width={"100%"}
+          firstLineWidth={"100%"}
+          lastLineWidth={"55%"}
+          animate={"shine"}
+          onReady={false}
+        />
+      </View>
+    </View>
+
+    <View style={styles.itemLoadingContentWrapper}>
+      <View style={styles.itemLoadingContentCenter}>
+        <PlaceholderLine />
+      </View>
+    </View>
   </View>
 );
 
+const SectionHeaderPlaceholder = (
+  <View style={styles.sectionHeaderWrapper}>
+    <View style={styles.sectionHeaderContent}>
+      <PlaceholderLine />
+    </View>
+  </View>
+);
 /**
  * A component to render messages with due_date in a agenda like form.
  */
 class MessageAgenda extends React.PureComponent<Props, State> {
+  private idTimeout?: number;
+  private idInterval?: number;
   // Ref to section list
   private sectionListRef = React.createRef<any>();
-  private idTimeoutProgress?: number;
-
   constructor(props: Props) {
     super(props);
     this.state = {
       itemLayouts: [],
       isLoadingProgress: false,
-      isFirstLoading: true
+      isFirstLoading: true,
+      isDeadlinesLoaded: false,
+      isLoadingComplete: false,
+      numMessagesToRender: 0
     };
-    this.loadMoreData = this.loadMoreData.bind(this);
+  }
+
+  public componentDidMount() {
+    // tslint:disable-next-line: no-object-mutation
+    this.idInterval = setInterval(() => {
+      if (!this.state.isDeadlinesLoaded) {
+        this.isDeadlinesLoadingComplete();
+      } else {
+        clearInterval(this.idInterval);
+      }
+    }, 300);
   }
 
   public componentWillUnmount() {
-    if (this.idTimeoutProgress !== undefined) {
-      clearTimeout(this.idTimeoutProgress);
+    // if a timeout is running we have to stop it
+    if (this.idTimeout !== undefined) {
+      clearTimeout(this.idTimeout);
+    }
+    if (this.idInterval !== undefined) {
+      clearInterval(this.idInterval);
     }
   }
 
-  // tslint:disable-next-line: cognitive-complexity
-  public componentDidUpdate(prevProps: Props) {
-    // Change status loading to show progress
+  public async componentDidUpdate() {
     if (
```

Comment:
```suggestion
    if (
      this.sectionListRef.current &&
      this.state.isFirstLoading &&
      this.state.isDeadlinesLoaded &&
      this.props.sections.length > 0
    )
```
please leave some comments to explain what is happening